operators/test/e2e/cmd/run/eventlog.go

```diff
@@ -0,0 +1,143 @@
+	el := &eventLogger{
+		eventInformer:         cache.NewSharedIndexInformer(eventWatch, &corev1.Event{}, kubePollInterval, cache.Indexers{}),
+		eventQueue:            workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "eck_e2e_events"),
+		interestingNamespaces: make(map[string]struct{}),
+		logFilePath:           logFilePath,
+	}
+
+	el.eventInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if key, err := cache.MetaNamespaceKeyFunc(obj); err == nil {
+				el.eventQueue.Add(key)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			if key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj); err == nil {
+				el.eventQueue.Add(key)
+			}
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			if key, err := cache.MetaNamespaceKeyFunc(newObj); err == nil {
+				el.eventQueue.Add(key)
+			}
+		},
+	})
+
+	// add all namespaces to interesting namespaces
+	s := struct{}{}
+	el.interestingNamespaces[testCtx.E2ENamespace] = s
+	el.interestingNamespaces[testCtx.GlobalOperator.Namespace] = s
+	for _, ns := range testCtx.NamespaceOperators {
+		el.interestingNamespaces[ns.Namespace] = s
+		el.interestingNamespaces[ns.ManagedNamespace] = s
+	}
+
+	return el
+}
+
+func (el *eventLogger) Start(stopChan <-chan struct{}) {
+	defer func() {
+		el.eventQueue.ShutDown()
+		runtime.HandleCrash()
+	}()
+
+	go el.eventInformer.Run(stopChan)
+
+	if !cache.WaitForCacheSync(stopChan, el.eventInformer.HasSynced) {
+		log.Error(errors.New("timed out waiting for cache to sync"), "Failed to sync event cache")
+		return
+	}
+
+	wait.Until(el.runEventProcessor, time.Second, stopChan)
+}
+
+func (el *eventLogger) runEventProcessor() {
+	logFile, err := os.Create(el.logFilePath)
+	if err != nil {
+		log.Error(err, "Failed to create event log", "path", el.logFilePath)
+		return
+	}
+	defer logFile.Close()
+	logWriter := json.NewEncoder(logFile)
+
+	for {
+		key, quit := el.eventQueue.Get()
+		if quit {
+			return
+		}
+
+		evtObj, exists, err := el.eventInformer.GetIndexer().GetByKey(key.(string))
+		if err != nil {
+			log.Error(err, "Failed to get event", "key", key)
+			return
+		}
+
+		if !exists {
+			continue
+		}
+
+		evt := evtObj.(*corev1.Event)
+		if el.isInterestingEvent(evt) {
+			logEntry := eventLogEntry{
+				Reason:    evt.Reason,
+				Message:   evt.Message,
+				Kind:      evt.InvolvedObject.Kind,
+				Name:      evt.InvolvedObject.Name,
+				Namespace: evt.InvolvedObject.Namespace,
+			}
+			if err := logWriter.Encode(logEntry); err != nil {
+				log.Error(err, "Failed to write event to event log", "event", evt)
+			}
+		}
+	}
+}
+
+// isInterestingEvent determines whether an event is worthy of logging.
+func (el *eventLogger) isInterestingEvent(evt *corev1.Event) bool {
+	// was the event generated by an object in one of the namespaces associated with this test run?
+	if _, exists := el.interestingNamespaces[evt.InvolvedObject.Namespace]; exists {
```

Comment:
I have noticed that "system events" are generated in the `default` namespace with a `Normal` type. 
e.g.:
```
- apiVersion: v1
  count: 1
  eventTime: null
  firstTimestamp: "2019-07-30T11:41:21Z"
  involvedObject:
    apiVersion: v1
    kind: PersistentVolume
    name: pvc-bb511d09-b2be-11e9-9340-42010a840052
    resourceVersion: "13388"
    uid: bee21d8b-b2be-11e9-9340-42010a840052
  kind: Event
  lastTimestamp: "2019-07-30T11:41:21Z"
  message: 'googleapi: Error 400: The disk resource ''projects/elastic-cloud-dev/zones/europe-west1-b/disks/gke-michael-dev-cluste-pvc-bb511d09-b2be-11e9-9340-42010a840052''
    is already being used by ''projects/elastic-cloud-dev/zones/europe-west1-b/instances/gke-michael-dev-cluster-default-pool-81094de3-wfhv'',
    resourceInUseByAnotherResource'
  metadata:
    creationTimestamp: "2019-07-30T11:41:21Z"
    name: pvc-bb511d09-b2be-11e9-9340-42010a840052.15b62c87aa23c3ea
    namespace: default
    resourceVersion: "634"
    selfLink: /api/v1/namespaces/default/events/pvc-bb511d09-b2be-11e9-9340-42010a840052.15b62c87aa23c3ea
    uid: f3ed7037-b2be-11e9-9340-42010a840052
  reason: VolumeDelete
  reportingComponent: ""
  reportingInstance: ""
  source:
    component: persistentvolume-controller
  type: Normal
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

Does it mean that they will not be logged ?

Reply:
Good find, that seems like it might be a bug in the PV controller though?

Reply:
Good catch. It's odd that the PV controller behaves that way. Looks like this particular occurrence will have to be added as a special case in order to capture and log it.

Reply:
Seems like it's intentional:
https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/pv_controller.go#L1170

Reply:
Mmm good find, does that mean we need to capture it?

Reply:
The comment in the source suggests that it is a transient problem that just needs some more time to recover. This might be an indication that our tests are not isolated enough because they are trying to reuse something that is in the process of being deleted. It's probably worth capturing the event so that it will remind us to find and fix the underlying issue.